datadog_checks_base/tests/base/checks/test_agent_check.py

```diff
@@ -914,3 +915,32 @@ def test_load_configuration_models(dd_run_check, mocker):
 
     assert check._config_model_instance is instance_config
     assert check._config_model_shared is shared_config
+
+
+@requires_py3
+def test_detect_typos_configuration_models(dd_run_check, mocker, caplog):
+    check_instance = {'endpoint': 'url', 'tags': ['foo:bar'], 'proxy': {'http': 'http://1.2.3.4:9000'}}
+    check_init_config = {'endpont': 'https://1.2.3.4:4242', 'tags': ['test'], 'proxy': 'test'}
+    check = AgentCheck('test', check_init_config, [check_instance])
+    check.check_id = 'test:123'
+
+    empty_config = {}
```

Comment:
Could you also test non-empty config that does not trigger a typo warning

Reply:
Yes definitely!

Reply:
Added some more tests. Let me know if you can think of other scenarios that should be tested!